yarn-project/simulator/src/public/enqueued_call_simulator.ts

```diff
@@ -122,33 +122,48 @@ export class EnqueuedCallSimulator {
   ): Promise<EnqueuedCallResult> {
     const pendingNullifiers = this.getSiloedPendingNullifiers(previousPublicKernelOutput);
     const startSideEffectCounter = previousPublicKernelOutput.endSideEffectCounter + 1;
+
+    const prevAccumulatedData =
+      phase === PublicKernelPhase.SETUP
+        ? previousPublicKernelOutput.endNonRevertibleData
+        : previousPublicKernelOutput.end;
+    const previousValidationRequestArrayLengths = PublicValidationRequestArrayLengths.new(
+      previousPublicKernelOutput.validationRequests,
+    );
+    const previousAccumulatedDataArrayLengths = PublicAccumulatedDataArrayLengths.new(prevAccumulatedData);
+
+    // If this is the first enqueued call in public, constants will be empty
+    // because private kernel does not expose them.
+    const constants = previousPublicKernelOutput.constants.clone();
```

Comment:
Interesting, why did you need to clone to fix the tests? Are these ever modified?

Reply:
I am modifying the constants in the next line to say `constants.globalVariables = this.globalVariables` because otherwise for the first enqueued call we give it empty globals. That's because the previous kernel's globals are empty if the previous kernel was private.

I think that by modifying this without cloning, I was modifying what was actually being sent as circuit inputs to the public kernel.